packages/core/src/browser/preferences/preference-configurations.ts

```diff
@@ -34,9 +36,16 @@ export class PreferenceConfigurations {
     @inject(ContributionProvider) @named(PreferenceConfiguration)
     protected readonly provider: ContributionProvider<PreferenceConfiguration>;
 
+    @inject(EnvVariablesServer)
+    protected readonly envServer: EnvVariablesServer;
+
+    /**
+     * Note: resolves to the name of the folders, (and not the path) where the preferences service is configured to look for settings.
+     */
     /* prefer Theia over VS Code by default */
-    getPaths(): string[] {
-        return ['.theia', '.vscode'];
+    async getPaths(): Promise<string[]> {
+        const configDirUri = await this.envServer.getConfigDirUri();
+        return [new URI(configDirUri).path.base, '.vscode'];
```

Comment:
I thin we mix things here, that's a directory for settings in workspace folders. In future it should stay `.theia`, regardless whether we want to have another name for user data folder.

Reply:
Good, that was my understanding too. So the initial changes were incorrect 👍

Reply:
Would it be fine to introduce another variable for it, different from `getConfigDirUri`?

Reply:
I would create a protected method here which returns with `'.theia.'` by default, clients can subclass it. But for me, it is not related to the env variable server. WDYT?

Reply:
sounds good, then we can unmake this method to be async and revert affected clients. We should document in CHANGELOG what should be done to change global and workspace config dirs, since we have 2 places at least.

Reply:
Good idea.

Reply:
Let's do it in a separate PR. I do not mind changing it but my feeling is that it has already caused some confusion. For me too.